app/src/test/java/org/mozilla/fenix/components/IntentProcessorTypeTest.kt

```diff
@@ -0,0 +1,114 @@
+        assertTrue(IntentProcessorType.NEW_TAB.shouldOpenToBrowser(intent))
+        assertFalse(IntentProcessorType.OTHER.shouldOpenToBrowser(intent))
+    }
+
+    @Test
+    fun `get type for normal intent processor`() {
+        val processor = testContext.components.intentProcessors.intentProcessor
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.NEW_TAB, type)
+        assertEquals(HomeActivity::class.java.name, type.activityClassName)
+    }
+
+    @Test
+    fun `get type for private intent processor`() {
+        val processor = testContext.components.intentProcessors.privateIntentProcessor
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.NEW_TAB, type)
+        assertEquals(HomeActivity::class.java.name, type.activityClassName)
+    }
+
+    @Test
+    fun `get type for custom tab intent processor`() {
+        val processor = testContext.components.intentProcessors.customTabIntentProcessor
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.EXTERNAL_APP, type)
+        assertEquals(ExternalAppBrowserActivity::class.java.name, type.activityClassName)
+    }
+
+    @Test
+    fun `get type for private custom tab intent processor`() {
+        val processor = testContext.components.intentProcessors.privateCustomTabIntentProcessor
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.EXTERNAL_APP, type)
+        assertEquals(ExternalAppBrowserActivity::class.java.name, type.activityClassName)
+    }
+
+    @Test
+    fun `get type for TWA intent processor`() {
+        val processor = testContext.components.intentProcessors.privateCustomTabIntentProcessor
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.EXTERNAL_APP, type)
+        assertEquals(ExternalAppBrowserActivity::class.java.name, type.activityClassName)
+    }
+
+    @Test
+    fun `get type for PWA intent processor`() {
+        val processor = testContext.components.intentProcessors.privateCustomTabIntentProcessor
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.EXTERNAL_APP, type)
+        assertEquals(ExternalAppBrowserActivity::class.java.name, type.activityClassName)
+    }
+
+    @Test
+    fun `get type for generic intent processor`() {
+        val processor = object : IntentProcessor {
+            override fun matches(intent: Intent) = true
+            override suspend fun process(intent: Intent) = true
+        }
+        val type = testContext.components.intentProcessors.getType(processor)
+
+        assertEquals(IntentProcessorType.OTHER, type)
+        assertEquals(HomeActivity::class.java.name, type.activityClassName)
+    }
+}
```

Comment:
Are we missing a new line at EOF?

Reply:
No, I'm pretty sure GitHub just hides the newline at the end. You can identify a missing newline with a 🚫